acousticbrainz/views/data.py

```diff
@@ -46,22 +47,41 @@ def view_high_level(mbid):
 def summary(mbid):
     lowlevel, highlevel, genres, moods, other = get_summary_data(mbid)
 
-    info = _get_track_info(mbid, lowlevel['metadata'])
+    info = None
+
+    if lowlevel == None: # No metadata in acousticbrainz.. try to fetch data from musicbrainz
+        info = _get_track_info(mbid)
+    else: # Try to get metadata from musicbrainz with second priority as the one retrieved from acoustic
+        info = _get_track_info(mbid, lowlevel['metadata'])
+
 
     # Tomahawk player stuff
-    if not ('artist' in lowlevel['metadata']['tags'] and 'title' in lowlevel['metadata']['tags']):
+    # Put additional check if the lowlevel has the metadata ????
+    # Why not directly take the data from info if present (doing that)
+    # Not checking if info is None or not because it is nver None
+
+    if not ('artist' in info and 'title' in info):
         tomahawk_url = None
     else:
         tomahawk_url = "http://toma.hk/embed.php?artist={artist}&title={title}".format(
             artist=quote_plus(info['artist'].encode("UTF-8")),
             title=quote_plus(info['title'].encode("UTF-8")))
 
-    return render_template("data/summary.html", lowlevel=lowlevel, highlevel=highlevel,
-                           genres=genres, moods=moods, other=other, mbid=mbid,
-                           info=info, tomahawk_url=tomahawk_url)
+    # Checking for what all information we have and rendering the pages accordingly..
 
+    if lowlevel and info: # When we have the data from the acousticbrainz available.. the happy condition..
+        return render_template("data/summary.html", lowlevel=lowlevel, highlevel=highlevel,
+                                genres=genres, moods=moods, other=other, mbid=mbid,
+                                info=info, tomahawk_url=tomahawk_url)
+    
+    elif info: # When there is only metadata... TODO: Move into the same template with if conditions...
+        return render_template("data/summary-metadata.html", info = info, tomahawk_url = tomahawk_url, mbid=mbid)
+    
+    else: # When there is no data
+        raise NotFound("MusicBrainz does not have data for this track. Please upload it!")
 
-def _get_track_info(mbid, metadata):
+
+def _get_track_info(mbid, metadata = None): # Defaulting the metadata to None
```

Comment:
Comment seems useless there since code is pretty obvious. This applies to some other comments you added as well. See http://blog.codinghorror.com/coding-without-comments/.